code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/dao/impl/DnsInfoDaoImpl.java

```diff
@@ -226,7 +240,9 @@ protected DnsEntryData map(List<Object> input) {
                         .and(clientNic.REMOVED.isNull())
                         .and(targetNic.REMOVED.isNull())
                         .and(clientServiceExposeMap.REMOVED.isNull())
-                        .and(targetServiceExposeMap.REMOVED.isNull()))
+                        .and(targetServiceExposeMap.REMOVED.isNull())
+                        .and(targetInstance.STATE.in(InstanceConstants.STATE_RUNNING,
+                                InstanceConstants.STATE_STARTING, InstanceConstants.STATE_RESTARTING)))
```

Comment:
I believe you, can remove. I just remember that for the standalone LB I used to include only Starting/Running containers as haproxy config, and for the restart case, the containers weren't registered just because at the moment of config generation they were at Restarting state. But may be for this use case the trigger point for config update - nic.activate - would ensure that instance is in Starting state. Let me know what you think